ci/scripts/r_docker_configure.sh

```diff
@@ -39,6 +39,19 @@ if [ "$RHUB_PLATFORM" = "linux-x86_64-fedora-clang" ]; then
   rm -rf $(${R_BIN} RHOME)/etc/Makeconf.bak
 fi
 
+# Special hacking to try to reproduce quirks on centos using non-default build
+# tooling.
+if [[ "$DEVTOOLSET_VERSION" -gt 0 ]]; then
+  if [ "`which dnf`" ]; then
+    dnf install -y centos-release-scl
+    dnf install -y "devtoolset-$DEVTOOLSET_VERSION"
+  else
+    yum install -y centos-release-scl
+    yum install -y "devtoolset-$DEVTOOLSET_VERSION"
+  fi
+  echo "source scl_source enable devtoolset-$DEVTOOLSET_VERSION" >> /etc/bashrc
```

Comment:
Maybe you can add `DEVTOOLSET_VERSION` to `environment` in docker-compose.yml and then (conditionally) `scl_source enable devtoolset-$DEVTOOLSET_VERSION"` in `r_test.sh`, rather than try to rely on bash profiles.